src/db.c

```diff
@@ -759,6 +764,31 @@ void scanGenericCommand(client *c, robj *o, unsigned long cursor) {
             }
         }
 
+        /* Filter an element if it isn't the type we want. */
+        if (!filter && o == NULL && typename){
+            robj* typecheck;
+            char *type;
+            typecheck = lookupKeyReadWithFlags(c->db, kobj, LOOKUP_NOTOUCH);
+            if (typecheck == NULL) {
+                type = "none";
+            } else {
+                switch(typecheck->type) {
+                case OBJ_STRING: type = "string"; break;
+                case OBJ_LIST: type = "list"; break;
+                case OBJ_SET: type = "set"; break;
+                case OBJ_ZSET: type = "zset"; break;
+                case OBJ_HASH: type = "hash"; break;
+                case OBJ_STREAM: type = "stream"; break;
+                case OBJ_MODULE: {
+                    moduleValue *mv = typecheck->ptr;
+                    type = mv->type->name;
```

Comment:
@gkorland Please have a look at how module type names are managed in `module.c`, it's not necessarily a null terminated.

Reply:
@yossigo I'll take a look

Reply:
In my defence, this is the same as how `typeCommand` does it https://github.com/antirez/redis/blob/unstable/src/db.c#L833-L835

Reply:
From a quick test, trying to pass a non null-terminated string will fail on module load as Redis has implicitly assumed that these will all be standard C strings - perhaps this should be stated explicitly